compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIEAssignTiles.cpp

```diff
@@ -73,15 +77,18 @@ LogicalResult findUsersInCoreAndAddTiles(
 
 /// Utility to clear non-local tile assignments.
 LogicalResult clearNonLocalTiles(RewriterBase &rewriter, Operation *op) {
-  op->walk([&](AMDAIE::LogicalObjectFifoFromMemrefOp objFifo) {
+  WalkResult res = op->walk([&](AMDAIE::LogicalObjFifoOpInterface objFifo) {
     if (objFifo.getMemorySpaceAsUInt() != 2) {
       rewriter.setInsertionPoint(objFifo);
       SmallVector<Value> tiles;
-      rewriter.replaceOpWithNewOp<AMDAIE::LogicalObjectFifoFromMemrefOp>(
-          objFifo, cast<LogicalObjectFifoType>(objFifo.getOutput().getType()),
-          objFifo.getMemref(), tiles);
+      if (failed(objFifo.replaceWithNewTiles(rewriter, tiles))) {
+        objFifo.emitOpError() << "could not replace its tiles.";
+        return WalkResult::interrupt();
+      }
```

Comment:
Is there any scenario where we might hit a `LogicalObjectFifoFromBuffersOp` in this flow or the one below ? If yes, a lit test might be warranted.
Else we could just do a `(void)(objFifo.replaceWithNewTiles(rewriter, tiles))`.

Reply:
Good point. I added a lit test for `logicalobjectfifo.from_buffers` and `logicalobjectfifo.placeholder`.